packages/transformers/less/src/LessTransformer.js

```diff
@@ -0,0 +1,92 @@
+// @flow strict-local
+
+import {Transformer} from '@parcel/plugin';
+import localRequire from '@parcel/local-require';
+
+export default new Transformer({
+  async getConfig({asset, resolve}) {
+    let config = await asset.getConfig(['.lessrc', '.lessrc.js'], {
+      packageKey: 'less'
+    });
+
+    if (config === null) {
+      config = {};
+    }
+
+    config.filename = asset.filePath;
+    config.plugins = [
+      ...(config.plugins || []),
+      urlPlugin({asset}),
+      resolvePathPlugin({asset, resolve})
+    ];
+
+    return config;
+  },
+
+  async transform({asset, config}) {
+    const less = await localRequire('less', asset.filePath);
+    const code = await asset.getCode();
+    let css;
+    try {
+      css = (await less.render(code, config)).css;
+    } catch (err) {
+      // For the error reporter
+      err.fileName = err.filename;
+      err.loc = {
+        line: err.line,
+        column: err.column
+      };
+      throw err;
+    }
+
+    asset.type = 'css';
+    asset.setCode(css);
+    asset.meta.hasDependencies = false;
+    return [asset];
+  }
+});
+
+function urlPlugin({asset}) {
+  return {
+    install(less, pluginManager) {
+      const visitor = new less.visitors.Visitor({
+        visitUrl(node) {
+          node.value.value = asset.addURLDependency(
+            node.value.value,
+            node.currentFileInfo.filename
+          );
+          return node;
+        }
+      });
+
+      visitor.run = visitor.visit;
+      pluginManager.addVisitor(visitor);
+    }
+  };
+}
+
+function resolvePathPlugin({asset, resolve}) {
+  return {
+    install(less, pluginManager) {
+      class LessFileManager extends less.FileManager {
+        supports() {
+          return true;
+        }
+
+        supportsSync() {
+          return false;
+        }
+
+        async loadFile(rawFilename, ...args) {
+          let filename = rawFilename;
+          if (filename[0] == '/' || filename[0] == '~') {
+            filename = await resolve(asset.filePath, filename);
```

Comment:
This should probably be in a try ... catch ... and check for the ~node_module syntax in the catch throwing a user friendly error.
Other than that this looks good.

Reply:
We should probably also support the `paths` feature of less config (#2653). Rather than reimplementing ourselves though, it would be good to fall back to super there instead. I think that would have to happen first, or perhaps instead of node_modules resolution. Webpack also does it that way: if there is a `paths` option, use the default resolver, otherwise do what you have here.

Reply:
This PR only uses the Parcel resolver if it starts with `/` or `~` and uses super.loadFile in all other cases which shouldn't break paths afaik

Reply:
oh I see... hmm what about node_modules resolution though? those wouldn't start with either of those chars.

Reply:
Good question, not sure how to solve that properly with paths

Reply:
Injecting the path of the `node_modules` folder to `paths` could work, but to do that we'll probably need access to the `NodeResolver` class, particularly its [`findNodeModulePath` method](https://github.com/arianon/parcel/blob/3d477d3fa0a37b89ff984dbf020b4b5cad110c70/packages/resolvers/default/src/DefaultResolver.js#L234-L271)

Reply:
Also @DeMoorJasper , I'm not entirely sure what error do you want caught in `loadFile`, care to elaborate?

Reply:
@arianon originally I thought you were gonna try to run it through Parcel's resolver first and than it would error if it can't find a file, but that wouldn't work for paths.
So your solution seems better. Although now node_modules is kinda an issue.

Reply:
Using your suggestion, but flipping it so the default Less resolver runs first, and then use the Parcel resolver when it fails, does the trick for both `paths` and `node_modules`